### PR TITLE
Test loading VP file

### PR DESCRIPTION
**Pull request by markdewing**
Follows #3640
Final part implementing the prototype in #3634

May also serve as a test for #3563 because the variational parameters are non-zero.

To avoid storing an HDF file in the repository, this test uses a script that converts from a text format to the HDF format.
The convert_vp_format.py script can also be useful for editing the variational parameter files (convert HDF to text, edit, and convert text to HDF)

The complex build stores the variational parameters as complex numbers, and expects to load them as complex numbers. Because of this, the variational parameter files will not be compatible across real and complex builds. The script deals with this by adding a "--complex" flag that will output the values in the complex number format.  The script can also be used to convert between the real and complex versions of the variational parameter files.  It will detect and read text or HDF files in either number format.  The "--complex" flag only affects output.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- No. This PR is up to date with current the current state of 'develop'
- N/A. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- N/A. Documentation has been added (if appropriate)

### CODE REVIEW

File: tests/molecules/He_param/CMakeLists.txt [hunk 1]
```diff
@@ -24,6 +24,41 @@ if(NOT QMC_CUDA)
...
+    qmc_run_and_check_custom_scalar(
+      BASE_NAME He_param_grad_load
+      BASE_DIR "${qmcpack_SOURCE_DIR}/tests/molecules/He_param"
+      PREFIX He_param_grad_load.param
+      INPUT_FILE He_param_grad_load.xml
+      PROCS 1
+      THREADS 16
+      SERIES 0
+      SCALAR_VALUES HE_BSPLINE_OPT_PARAM)
+
```
Comment (ye-luo) [inline tests/molecules/He_param/CMakeLists.txt:58]: I'm a bit lost. Where is h5diff being triggered?

Comment (markdewing) [reply tests/molecules/He_param/CMakeLists.txt:58]: The test compares the computed parameter gradients.    So h5diff is not used.
The test sets up the wavefunction and loads from the VP file, computes parameter gradients, and then compares those gradients to expected values.

Comment (markdewing) [reply tests/molecules/He_param/CMakeLists.txt:58]: There's no test that runs the optimizer and outputs VP files, then runs a comparison on the VP files.  Maybe there should be.

Comment (ye-luo) [reply tests/molecules/He_param/CMakeLists.txt:58]: I see. txt is converted to vp.h5 and then loaded by QMCPACK.

File: tests/molecules/He_param/CMakeLists.txt [hunk 2]
```diff
@@ -24,6 +24,41 @@ if(NOT QMC_CUDA)
...
+    set(TDIR "${qmcpack_BINARY_DIR}/tests/molecules/He_param/He_param_grad_load-1-16/")
+    file(MAKE_DIRECTORY ${TDIR})
+    execute_process(COMMAND ${qmcpack_SOURCE_DIR}/tests/molecules/He_param/convert_vp_format.py ${SDIR}/he_vp_opt.txt ${complex_flag} -o ${TDIR}/he_vp_opt.h5)
+
+    list(APPEND HE_BSPLINE_OPT_PARAM jud_0  0.00000124  0.0014) # scalar name, value, error
+    list(APPEND HE_BSPLINE_OPT_PARAM jud_1 -0.000273    0.00097)
+    list(APPEND HE_BSPLINE_OPT_PARAM jud_2 -0.000181    0.00082)
+    list(APPEND HE_BSPLINE_OPT_PARAM jud_3  0.0004463   0.000058)
+
+    qmc_run_and_check_custom_scalar(
```
[the remark this answers is not in the document]
Comment (markdewing) [reply tests/molecules/He_param/CMakeLists.txt:49]: It might be possible to replace `qmc_run_and_check` completely with `qmc_run_check_custom_scalar`.

[the remark this answers is not in the document]
Comment (ye-luo) [reply tests/molecules/He_param/CMakeLists.txt:49]: #3663

File: tests/molecules/He_param/CMakeLists.txt [hunk 3]
```diff
@@ -24,6 +24,41 @@ if(NOT QMC_CUDA)
     0
     SCALAR_VALUES
     HE_BSPLINE_PARAM)
+
+
+  #  Test loading from variational parameter file
+  check_python_reqs(h5py he_param_h5 add_h5_tests)
```
Comment (ye-luo) [inline tests/molecules/He_param/CMakeLists.txt:30]: I think we can do better to reduce checks #3662

**Review by ye-luo — APPROVED** [review]

**Comment by prckent** [comment]
Suggestion: version in the parameter file is the version of QMCPACK?

**Review by prckent — APPROVED** [review]
After reply to my comment

**Comment by markdewing** [comment]
The version number is a way to determine if the format changes.   If there is no version number, we'll have to use a different mechanism to determine if the version changes.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
@prckent the previous Gcov failure was likely a glitch in the connection between Gcov and github.
